docs/reference/contrib/frontendcache.rst

```diff
@@ -170,11 +176,9 @@ Let's take the the above BlogIndexPage as an example. We need to register a sign
 
 
     def blog_page_changed(blog_page):
-        # Find all the live BlogIndexPages that contain this blog_page
-        for blog_index in BlogIndexPage.objects.live():
-            if blog_page in blog_index.get_blog_items().object_list:
-                # Purge this blog index
-                purge_page_from_cache(blog_index)
+        # Find the blog index and purge it. Assuming there is just one on the site.
```

Comment:
The existing docs ("Find all the live BlogIndexPages") and code seem more consistent with the [logic below](https://github.com/wagtail/wagtail/pull/3824/files#diff-ce9d3dbe263a870a67b78eef6a26e59aR231) and the text above ("This signal handler should call the ``purge_page_from_cache`` function on all BlogIndexPages that contain the BlogPage being updated/deleted.").

Reply:
I might have been both trying to avoid providing inefficient code in the example and introducing batch invalidation early.

After some thought, I eventually opted for ditching ``purge_page_from_cache`` entirely from the docs and suggest people always use ``PurgeBatch``. I think it's simpler this way as having two different APIs do the same thing could be confusing, and ``PurgeBatch`` is the more flexible of the two.

I'm wondering if we should add documentation for them anyway, as there is code out there using them. It might be a good idea to deprecate them, but I don't think they're a problem for maintenance anyway. Do you have any thoughts on this?

Reply:
@kaedroho it might be nice to keep the docs mention of `purge_page_from_cache` as it's a simpler way to purge a single page, as opposed to creating a `PurgeBatch` and we'd want to keep it around for backwards compatibility (at least for now).